### PR TITLE
Add timeseries plots to upgrade measure postprocessing

### PR DESCRIPTION
**Pull request by ChristopherCaradonna**
Pull request overview
---------------------

Adds the option to produce several timeseries plots comparing the baseline to upgrades. Plots include peak week by season, seasonal averages by day type, and seasonal averages by end use.

### Pull Request Author: Chris CaraDonna

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes apptainer image, buildstock batch, dependencies, continuous integration tests)
 - [ ] Sampling
 - [ ] Workflow Measures
 - [x] Upgrade Measures
 - [ ] Reporting Measures
 - [x] Postprocessing

Author pull request checklist:

 - [ ] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [ ] Added tests for new measures
 - [ ] Updated measure .xml(s)
 - [ ] Register values added to `comstock_column_definitions.csv`
 - [ ] Both `options_lookup.tsv` files updated
 - [x] 10k+ test run
 - [ ] Change documentation written
 - [ ] Measure documentation written
 - [ ] ComStock documentation updated
 - [ ] Changes reflected in example `.yml` files
 - [ ] Changes reflected in `README.md` files
 - [ ] Added 'See ComStock License' language to first two lines of each code file
 - [ ] Implements corresponding measure tests and indexing path in `test/measure_tests.txt` or/and `test/resource_measure_tests.txt`
 - [ ] All new and existing tests pass the CI

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] Make sure the newly added measures has been added with tests and indexed properly
 - [ ] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

### CODE REVIEW

File: postprocessing/compare_upgrades.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
```
Comment (asparke2) [inline postprocessing/compare_upgrades.py]: Should delete this file and just keep the .template version

File: postprocessing/comstockpostproc/comstock_measure_comparison.py [hunk 2]
```diff
@@ -96,24 +99,30 @@ def __init__(self, comstock_object, image_type='jpg', name=None, make_comparison
...
-        self.plot_eui_histograms_by_building_type(df, column_for_grouping, color_map, output_dir)
-        self.plot_eui_boxplots_by_building_type(df, column_for_grouping, color_map, output_dir)
-        self.plot_measure_savings_distributions_enduse_and_fuel(df, output_dir)
-        self.plot_measure_savings_distributions_by_building_type(df, output_dir)
-        self.plot_measure_savings_distributions_by_climate_zone(df, output_dir)
-        self.plot_measure_savings_distributions_by_hvac_system_type(df, output_dir)
-        self.plot_qoi_timing(df, column_for_grouping, color_map, output_dir)
-        self.plot_qoi_max_use(df, column_for_grouping, color_map, output_dir)
-        self.plot_qoi_min_use(df, column_for_grouping, color_map, output_dir)
+        #self.plot_energy_by_enduse_and_fuel_type(df, column_for_grouping, color_map, output_dir)
```
Comment (asparke2) [inline postprocessing/comstockpostproc/comstock_measure_comparison.py]: Think these should not be commented out

File: postprocessing/comstockpostproc/plotting_mixin.py [hunk 3]
```diff
@@ -2110,3 +2113,629 @@ def plot_load_duration_curve(self, df, region, building_type, color_map, output_
...
+
+        # concatinate and combine upgrade data
+        dfs_upgrade_combined = pd.concat(dfs_up, join='outer', ignore_index=True)
+        dfs_upgrade_combined = dfs_upgrade_combined.groupby(['time'], as_index=False)[dfs_upgrade_combined.loc[:, dfs_upgrade_combined.columns.str.contains('_kwh')].columns].sum()
+        dfs_upgrade_combined[self.UPGRADE_NAME] = upgrade_name
+
+        return dfs_base_combined, dfs_upgrade_combined
+
+    # plot
+    order_list = [
```
Comment (asparke2) [inline postprocessing/comstockpostproc/plotting_mixin.py:2704]: There are 3 copies of this enduse list and corresponding color map in this file; is there a way to make this a Constant  that returns these things for reuse?

**Review by asparke2 — CHANGES_REQUESTED** [review]
@ChristopherCaradonna looks good, 2 comments necessary to address, 1 that's optional (moving the color lists).

**Review by rHorsey — APPROVED** [review]
lgtm - thanks so much @ChristopherCaradonna

**Pull request merged**